### PR TITLE
load a single layer via a query param

### PR DESCRIPTION
**Pull request by camallen**
Specify layer query param to load a specific layer from the API and then zoom.

### CODE REVIEW

File: js/maps.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { queryParams } from './queryParams.js'
```
Comment (eatyourgreens) [inline js/maps.js:1]: I've been avoiding `import` because it may not work in the browser.

File: js/maps.js [hunk 2]
```diff
@@ -220,13 +209,32 @@ function zoomToFit() {
...
+  getLayerFunc = 'pendingLayers';
+} else if (layer) {
+  getLayerFunc = 'layer';
+  zoomFunc = zoomToFit;
+}
+
+API[getLayerFunc](eventName, layer)
   .then(buildLayersMenu)
-  .then(renderMapAndFit);
+  .then(renderMap)
```
Comment (eatyourgreens) [inline js/maps.js]: I don't think `renderMap` is a promise, so I think the next `then` block will run immediately.

File: js/maps.js [hunk 3]
```diff
@@ -220,13 +209,32 @@ function zoomToFit() {
...
+  .then(zoomFunc);
```
Comment (eatyourgreens) [inline js/maps.js]: What you want to do here, I think, is pass `zoomFunc` to Papaparse as the `complete` callback in the parser config. https://www.papaparse.com/docs#config

Maybe pass it as an argument to the `renderMap` function?

File: js/maps.js (hunk 2 above)
Comment (eatyourgreens) [reply js/maps.js]: Ignore that, [MDN says it has good support now](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#Browser_compatibility).

File: js/maps.js (hunk 1 above)
Comment (eatyourgreens) [reply js/maps.js:1]: Ignore that, [MDN says it has good support now](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#Browser_compatibility).

Comment (camallen) [reply js/maps.js:1]: as the comment about good support on this import statement? It looked like it had ok support when i went this way - can remove if needed.

File: js/maps.js (hunk 3 above)
Comment (camallen) [reply js/maps.js]: I've done it slightly differently and promisified the `renderMap()` function and used the promise resolve as the callback to the parser config. 

I'm not sure if passing the callback function to renderMap is more optimal and clearer / cleaner in the code? Happy to change.

Comment (eatyourgreens) [reply js/maps.js]: Sounds good to me. `renderMap()` is set up to handle an array of layers, so you might need to use `Promise.all()`. I can never remember how iteration works for `nodeLists` returned from query selectors, which are slightly different from arrays.

File: js/maps.js [hunk 4]
```diff
@@ -179,32 +165,38 @@ function toggleLayer(event) {
...
-    .forEach(function (node) {
-      const url = node.value;
-      if (HEATMAPS[url]) {
-        const heatmapData = filteredMapData(HEATMAP_DATA[url]);
-        HEATMAPS[url].setData(heatmapData);
-        HEATMAPS[url].setMap(map);
-      } else {
-        readMapFile(url);
-      }
+  return new Promise(function (resolve, reject) {
```
Comment (eatyourgreens) [inline js/maps.js]: If there's more than one selected checkbox, I think this will resolve for the first selected layer.

File: js/maps.js [hunk 5]
```diff
@@ -179,32 +165,38 @@ function toggleLayer(event) {
...
+    urls.forEach(function (url) {
+      HEATMAPS[url] && HEATMAPS[url].setMap(null);
     })
+    MAP_SELECT.querySelectorAll('input:checked')
+      .forEach(function (node) {
+        const url = node.value;
+        if (HEATMAPS[url]) {
+          const heatmapData = filteredMapData(HEATMAP_DATA[url]);
+          HEATMAPS[url].setData(heatmapData);
+          HEATMAPS[url].setMap(map);
```
Comment (eatyourgreens) [inline js/maps.js]: If the layer is already cached in `HEATMAPS[url]`, then the promise doesn't resolve. Does that cause problems showing/hiding layers?

**Review by eatyourgreens — APPROVED** [review]

**Pull request merged**